### DISCUSSION TITLE
Monorepo

### DISCUSSION THREAD

**Discussion by zoran995** [TerriaJS/terriajs#6450] [Ideas]
Thinking about this a lot lately and would like to hear the TerriaJS team opinion on it. What do you think about switching terriajs to the monorepo (in v9/v10 or even later), i.e. traits can be their own package `@terriajs/traits` (so we can use them in the backend without entire terriajs), catalog items can also be extracted as proposed in #6227

**Comment by staffordsmith83** [comment]
This sounds useful - "traits can be their own package @terriajs/traits (so we can use them in the backend without entire terriajs)"

**Comment by tephenavies** [comment]
Breaking down terriajs into parts may be useful, but I think it is less important than making terriajs a proper library (builds to js, typings & assets and can be imported into e.g. TerriaMap, rather than only building as part of TerriaMap) and having a proper plugin interface.

terriajs should be usable in the backend as is.
